sys-utils/libcmt/Makefile

```diff
@@ -81,19 +74,62 @@ $(libcmt_SO): $(libcmt_OBJ)
 	$(TARGET_CC) -shared -o $@ $^
 
 libcmt: $(libcmt_LIB) $(libcmt_SO)
+install-run: $(libcmt_SO)
+	mkdir -p $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib
+	cp -f $(libcmt_SO) $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib
+
 install: $(libcmt_LIB) $(libcmt_SO) build/ffi.h
 	mkdir -p $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib
-	cp -f $(libcmt_LIB) $(libcmt_SO) $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib
+	cp -f $(libcmt_SO) $(libcmt_LIB) $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib
 	mkdir -p $(TARGET_DESTDIR)$(TARGET_PREFIX)/include/libcmt/
 	cp -f include/libcmt/*.h $(TARGET_DESTDIR)$(TARGET_PREFIX)/include/libcmt/
 	cp -f build/ffi.h $(TARGET_DESTDIR)$(TARGET_PREFIX)/include/libcmt/
 	mkdir -p $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib/pkgconfig
-	sed -e 's|@ARG_PREFIX@|$(TARGET_PREFIX)|g' src/libcmt.pc > $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib/pkgconfig/libcmt.pc
-
-debian-package: install
-	mkdir -p $(TARGET_DESTDIR)/DEBIAN
-	sed 's|ARG_VERSION|$(LIBCMT_VERSION)|g;' tools/template/cross-control.template > $(TARGET_DESTDIR)/DEBIAN/control
-	dpkg-deb -Zxz --root-owner-group --build $(TARGET_DESTDIR) $(LIBCMT_DEB_FILENAME)
+	sed -e 's|@PREFIX@|$(TARGET_PREFIX)|g' \
+	    tools/libcmt.pc.in > $(TARGET_DESTDIR)$(TARGET_PREFIX)/lib/pkgconfig/libcmt.pc
+
+# requires either install-run or install
+debian-package:
+	mkdir -p $(dir $(DEB_FILENAME))
+	dpkg-deb -Zxz --root-owner-group --build $(TARGET_DESTDIR) $(DEB_FILENAME)
+
+$(TARGET_DESTDIR)/DEBIAN/control: tools/control.in
+	mkdir -p $(@D)
+	sed -e 's|@PACKAGE@|$(ARG_PACKAGE)|' \
+	    -e 's|@VERSION@|$(ARG_VERSION)|' \
+	    -e 's|@ARCHITECTURE@|$(ARG_ARCHITECTURE)|' \
+	    -e 's|@SECTION@|$(ARG_SECTION)|' \
+	    -e 's|@PROVIDES@|$(ARG_PROVIDES)|' \
+	    -e 's|@DESCRIPTION@|$(ARG_DESCRIPTION)|' \
+	    $< > $@
+
+# debian package variants
+libcmt-v$(ARG_VERSION).deb:
+	$(MAKE) $(TARGET_DESTDIR)/DEBIAN/control \
+	        ARG_PACKAGE=libcmt \
+	        ARG_ARCHITECTURE=riscv64 \
+	        ARG_PROVIDES=libcmt \
+	        ARG_SECTION=libs \
+	        ARG_DESCRIPTION="Cartesi Machine Tools"
+	$(MAKE) debian-package DEB_FILENAME=build/deb/$@
+
+libcmt-dev-v$(ARG_VERSION).deb:
+	$(MAKE) $(TARGET_DESTDIR)/DEBIAN/control \
+	        ARG_PACKAGE=libcmt-dev \
+	        ARG_ARCHITECTURE=any \
```

Comment:
I am having problems when installing it:
```
root@f1160b50b687:/mnt# dpkg -i libcmt-dev-v0.15.0.deb 
dpkg: error processing archive libcmt-dev-v0.15.0.deb (--install):
 package architecture (any) does not match system (riscv64)
Errors were encountered while processing:
 libcmt-dev-v0.15
```

Reply:
Install with `apt` instead.
```
docker run --rm -it --platform linux/riscv64 riscv64/ubuntu
$ apt install ./libcmt/deb/libcmt-v0.15.0.deb
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'libcmt' instead of './libcmt/deb/libcmt-v0.15.0.deb'
The following NEW packages will be installed:
  libcmt
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/40.4 kB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 /home/m/dev/cartesi/sdk/tools/libcmt/deb/libcmt-v0.15.0.deb libcmt riscv64 0.15.0 [40.4 kB]
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package libcmt.
(Reading database ... 4384 files and directories currently installed.)
Preparing to unpack .../libcmt/deb/libcmt-v0.15.0.deb ...
Unpacking libcmt (0.15.0) ...
Setting up libcmt (0.15.0) ...
```